app/javascript/packages/document-capture/components/full-screen.jsx

```diff
@@ -1,81 +1,139 @@
-import { useRef, useEffect, useCallback } from 'react';
+import { useRef, useEffect } from 'react';
+import { createPortal } from 'react-dom';
 import { createFocusTrap } from 'focus-trap';
 import useI18n from '../hooks/use-i18n';
 import useAsset from '../hooks/use-asset';
 
+/** @typedef {import('focus-trap').FocusTrap} FocusTrap */
 /** @typedef {import('react').ReactNode} ReactNode */
 
 /**
- * @typedef FullScreenProps
- *
- * @prop {()=>void=} onRequestClose Callback invoked when user initiates close intent.
- * @prop {ReactNode} children       Child elements.
+ * @typedef {()=>void} RequestCloseCallback
  */
 
 /**
- * Number of active instances of FullScreen currently mounted, used in determining when overlay body
- * class should be added or removed.
+ * @typedef FullScreenProps
  *
- * @type {number}
+ * @prop {RequestCloseCallback=} onRequestClose Callback invoked when user initiates close intent.
+ * @prop {string} label Accessible label for modal.
+ * @prop {ReactNode} children Child elements.
  */
-let activeInstances = 0;
 
 /**
- * @param {FullScreenProps} props Props object.
+ * @param {React.MutableRefObject<HTMLElement?>} containerRef
+ * @param {RequestCloseCallback} onRequestClose
  */
-function FullScreen({ onRequestClose = () => {}, children }) {
-  const { t } = useI18n();
-  const { getAssetPath } = useAsset();
-  const trapRef = useRef(/** @type {?import('focus-trap').FocusTrap} */ (null));
+function useFocusTrap(containerRef, onRequestClose) {
```

Comment:
The intention with the hooks refactoring here is to try to make these behaviors more self-documenting and avoid littering each's dependencies at the top-level of `FullScreen`. Especially coming back to this several months later, I find the standard approach of `useEffect(someFunc)` to be non-obvious as far as what that behavior is intended to accomplish.

I tried to design each as though they could be used in any component, even if not going so far as to extract it to be reusable. However, on reflection, I think it could make sense to do just that, since we have some conventions already around defining hooks, and it'd help reduce the size of this file.